app/services/waste_notification.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+class WasteNotification
+  attr_accessor :client, :message_options, :check_date, :waste_types
+
+  def initialize(check_date = Date.today)
+    @check_date = check_date
+  end
+
+  def send_notifications
+    all_waste_registrations = Waste::DeviceRegistration.all
+    # Waste::LocationType
+
+    # Load all possible addresses for registered push notifications
+    possible_addresses = Address.joins(:waste_location_types).where(
+      street: all_waste_registrations.pluck(:street),
+      city: all_waste_registrations.pluck(:city),
+      zip: all_waste_registrations.pluck(:zip)
+    )
+    # possible_addresses
+
+    # Load all PickupTimes of next couple days
+    number_of_days = all_waste_registrations.pluck(:notify_days_before).uniq
+    date_span = dates_to_check(number_of_days)
+    pickup_times = Waste::PickUpTime.where(pickup_date: date_span)
+
+    # load all Waste::LocationType in given addreses and timespan
+    waste_pickup_times = Waste::PickUpTime
+                             .includes(:waste_location_type)
+                             .where(id: pickup_times.map(&:id))
+                             .where(waste_location_types: { address_id: possible_addresses.map(&:id) })
+
+    all_waste_registrations.each do |registration_to_check|
```

Comment:
hier muss ich mal wieder benennungen anmerken. wieso wird nicht das schema "mehrzahl - einzahl" genutzt?
`all_waste_registrations` => `waste_registrations`
`registration_to_check` => `waste_registration`

oder ist es was anderes? bei einer `registration_to_check` ist ja im späteren verlauf nicht mehr klar, was das ist.
eigentlich sollte es sogar `waste_device_registrations` und `waste_device_registration` sein, da das model ja `Waste::DeviceRegistration` lautet. es reichtja einmal über benennungen nachzudenken, für die models, bevor man im code loslegt und diese dann durchzieht ohne erneut drüber nachdenken zu müssen!?

Reply:
ruby macht ja zum beispiel auch nicht ohne grund aus `Waste::LocationType` bei model-verknüpfungen wie `waste_pickup_time.waste_location_type` per convention das draus: `waste_location_type`

Reply:
der Modelname alleine schafft hier aber keine Klarheit. Es geht darum das alle Registrierungen geprüft werden müssen und in der Schleife für jede `registration_to_check` die waste_pickup_times geprüft werden.

Reply:
was ist der unterschied zwischen "jede waste registration" und "jede registration to check"?